File: frontend/src/modules/mod_test/mod_test.js
import api from '@/core/api.js'

/**
 * 【全栈样板逻辑层】
 * 展示前端如何连接后端 API，并处理响应状态。
 */
export function setupTestModule(store, actions) {
    return {
        // 核心功能：全栈探测数据属性
        async probeDataStructure() {
            if (!store.currentDataFile) {
                actions.openAlert("提示", "请先注入或上传一份数据集！");
                return;
            }

            actions.addLog("🔄 全栈探测请求已发送至后端...", "info");
            store.testLoading = true;

            try {
                // 调用后台刚刚编写好的接口
                const res = await api.post('/api/test/probe', {
                    filename: store.currentDataFile
                });

                if (res.data.status === 'success') {
                    // 将后端返回的统计数据注入全局状态
                    store.testResult = res.data.data;
                    actions.addLog("✅ 后端透视成功！数据结构已解析完毕。", "success");
                } else {
                    actions.addLog(`❌ 后端处理失败: ${res.data.message}`, "error");
                }
            } catch (e) {
                console.error(e);
                actions.addLog("🔥 网络连接异常，无法触达后端接口。", "error");
            } finally {
                store.testLoading = false;
            }
        }
    };
}

File: backend/modules/mod_test/mod_test.py
import os
import pandas as pd
from flask import Blueprint, request, jsonify
from core.config import UPLOAD_FOLDER
from core.utils import read_df

bp = Blueprint('test_module', __name__)

@bp.route('/api/test/probe', methods=['POST'])
def probe_data():
    """
    【全栈样板：数据探测器】
    演示接收前端文件名 -> 读取物理文件 -> 执行算法逻辑 -> 返回格式化结果
    """
    try:
        data = request.json
        filename = data.get('filename')
        
        if not filename:
            return jsonify({"status": "error", "message": "缺失文件名"}), 400
            
        file_path = os.path.join(UPLOAD_FOLDER, filename)
        if not os.path.exists(file_path):
            return jsonify({"status": "error", "message": "后端找不到指定物理文件"}), 404
            
        # 使用系统内置工具读取数据
        df = read_df(file_path)
        
        # 执行简单的分析逻辑：探测各列的统计学属性
        summary = []
        for col in df.columns:
            dtype = str(df[col].dtype)
            summary.append({
                "name": col,
                "type": "数值型" if "int" in dtype or "float" in dtype else "文本/类别型",
                "null_count": int(df[col].isnull().sum()),
                "unique_values": int(df[col].nunique())
            })
            
        return jsonify({
            "status": "success",
            "data": {
                "columns_summary": summary,
                "total_rows": len(df),
                "total_cols": len(df.columns)
            }
        })
    except Exception as e:
        return jsonify({"status": "error", "message": f"后端透视失败: {str(e)}"}), 500
